lib/amqp/Publication.js

```diff
@@ -46,13 +46,17 @@ function Publication(getChannelFn, publishFn, vhost, config) {
 
     function _publish(buffer, publishConfig, next) {
         var emitter = new EventEmitter()
-        getChannelFn(vhost, emitter, config, function(err, channel) {
-            if (err) return emitter.emit('error', err)
-            channel.on('return', emitter.emit.bind(emitter, 'return'))
+        var errorHandler = handleChannelError.bind(null, emitter, config)
+        var returnHandler = emitter.emit.bind(emitter, 'return')
+        borrowChannelFn(vhost, errorHandler, returnHandler, function(err, channel) {
+            if (err) {
+                if (channel) returnChannelFn(vhost, channel, errorHandler, returnHandler)
```

Comment:
Belt and braces. But by just looking at the functions we're calling here you can't ever get both. So I'll get rid of it.